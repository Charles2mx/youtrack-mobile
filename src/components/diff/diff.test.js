import React from 'react';

import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import Diff from './diff';

describe('<Diff/>', () => {

  let wrapper;
  let text1;
  let text2;

  describe('Render', () => {
    text1 = 'ABCy';
    text2 = 'xABC';
    beforeEach(() => {
      wrapper = doShallow(text1, text2);
    });

    it('should match a snapshot', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render component', () => {
      expect(findByTestId('diff')).toHaveLength(1);
    });

    it('should render difference', () => {
      expect(findByTestId('diffInsert')).toHaveLength(1);
      expect(findByTestId('diffDelete')).toHaveLength(1);
      expect(findByTestId('diffEqual')).toHaveLength(1);
    });
  });


  function findByTestId(testId) {
    return wrapper && wrapper.find({testID: testId});
  }

  function doShallow(text1, text2) {
    return shallow(
      <Diff text1={text1} text2={text2}/>
    );
  }
});
