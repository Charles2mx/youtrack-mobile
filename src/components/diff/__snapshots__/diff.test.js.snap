// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Diff/> Render should match a snapshot 1`] = `
<Text
  testID="diff"
>
  <Text
    key="0"
    style={
      Object {
        "backgroundColor": "#E6FFE6",
      }
    }
    testID="diffInsert"
  >
    x
  </Text>
  <Text
    key="1"
    style={Object {}}
    testID="diffEqual"
  >
    ABC
  </Text>
  <Text
    key="2"
    style={
      Object {
        "backgroundColor": "#FFE6E6",
      }
    }
    testID="diffDelete"
  >
    y
  </Text>
</Text>
`;
